api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -98,6 +100,79 @@ public void validate_shouldRejectAVisitIfItHasFewerThanMinOccursOfAnAttribute()
 	
 	/**
 	 * @see VisitValidator#validate(Object,Errors)
+	 * @verifies reject a visit if it has more than max occurs of an attribute when expected exactly one
+	 */
+	@Test
+	public void validate_shouldRejectAVisitIfItHasMoreThanMaxOccursOfAnAttributeWhenExpectedExactlyOne() throws Exception {
+		executeDataSet("org/openmrs/validator/include/VisitValidatorTest-additionalAttributeType1.xml");
+		
+		Visit visit = makeVisit();
+		visit.addAttribute(makeAttribute("one"));
+		visit.addAttribute(makeAttribute("two"));
+		visit.addAttribute(makeAttribute("three"));
+		visit.addAttribute(makeAttribute("four"));
+		try {
+			ValidateUtil.validate(visit);
+			fail();
+		}
+		catch (ValidationException e) {
+			Errors errors = e.getErrors();
+			assertEquals(1, errors.getErrorCount());
```

Comment:
Did you intentionally decide not to use @Test(expected = ValidationException.class)
or http://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html

http://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests